src/server/models/Game.ts

```diff
@@ -0,0 +1,30 @@
+export const games: GameType[] = [];
+
+export class Game {
+  room: string;
+  users: UserType[];
+  
+  constructor(game: GameType) {
+    const { room, users } = game;
+    this.users = users;
+    this.room = room;
+    games.push(this);
+  }
+
+  getUsers(): UserType[] {
+    return this.users;
+  }
+
+  // If game exists then room name is taken and user can only join the game
+  // Should it be not a method of this class?
```

Comment:
Should we create 3 classes

Game 1 ---- * Room 1 ------- * User

Reply:
3 or more
what "----- *" means?

Reply:
1 ----- * is relationship one to many 

one game has many rooms

Reply:
I'm not sure about "1 game many rooms", I thought 1 game = 1 room

Can you explain?

Reply:
Game here is like the main controller. they can be many rooms playing in the same time

Reply:
ok, I thought about having Games = Game[], but we can do Game = Room[]